eos_data_distribution/ndn/http.py

```diff
@@ -99,25 +111,31 @@ def _got_stream(self, msg, task, data):
 
     def _got_buf(self, data, buf):
         data.setContent(buf)
-        self.sendFinish(data)
+        self.emit('data', data)
+        logger.debug ('getter: got buf')
 
+class Producer(chunks.Producer):
+    def __init__(self, name, url, session=None, *args, **kwargs):
+        super(Producer, self).__init__(name, cost=DEFAULT_COST, *args, **kwargs)
+        self._getter = Getter(url, session=session, chunk_size=self.chunk_size)
```

Comment:
Should `assert(chunk_size > 0)` somewhere (possibly in the parent constructor?), because we divide by the chunk size in `_get_final_segment()`.